v8engine/src/main/java/cm/aptoide/pt/v8engine/view/account/user/ManageUserPresenter.java

```diff
@@ -22,52 +22,99 @@
   private final AptoideAccountManager accountManager;
   private final ThrowableToStringMapper errorMapper;
   private final FragmentNavigator fragmentNavigator;
+  private final ManageUserFragment.ViewModel userData;
+  private final boolean isEditProfile;
 
   public ManageUserPresenter(ManageUserView view, CrashReport crashReport,
       AptoideAccountManager accountManager, ThrowableToStringMapper errorMapper,
-      FragmentNavigator fragmentNavigator) {
+      FragmentNavigator fragmentNavigator, ManageUserFragment.ViewModel userData,
+      boolean isEditProfile) {
     this.view = view;
     this.crashReport = crashReport;
     this.accountManager = accountManager;
     this.errorMapper = errorMapper;
     this.fragmentNavigator = fragmentNavigator;
+    this.userData = userData;
+    this.isEditProfile = isEditProfile;
   }
 
   @Override public void present() {
+    handleSaveDataClick();
+    handleCancelClick();
+    handleSelectImageClick();
+    onViewCreatedLoadUserData();
+  }
 
-    Observable<Void> handleSaveDataClick = view.saveUserDataButtonClick()
-        .doOnNext(__ -> view.showProgressDialog())
-        .flatMap(userData -> saveUSerData(userData).observeOn(AndroidSchedulers.mainThread())
-            .doOnCompleted(() -> view.dismissProgressDialog())
-            .doOnCompleted(() -> sendAnalytics(userData.hasImage()))
-            .doOnCompleted(() -> navigateAway(userData.isEditProfile()))
-            .onErrorResumeNext(err -> handleSaveUserDataError(err, userData.isEditProfile()))
-            .toObservable())
-        .retry()
-        .map(__ -> null);
+  @Override public void saveState(Bundle state) {
+    // does nothing
+  }
 
-    Observable<Void> handleCancelClick = view.cancelButtonClick()
-        .doOnNext(__ -> navigateBack());
+  @Override public void restoreState(Bundle state) {
+    // does nothing
+  }
 
-    Observable<Void> handleSelectImageClick = view.selectUserImageClick()
-        .retry()
-        .doOnNext(__ -> view.showLoadImageDialog());
+  private void onViewCreatedLoadUserData() {
+    view.getLifecycle()
```

Comment:
it is a single, so when one result returns it gets subscribed. will add composition to view destruction even in case that happens before the result though.